browser/extensions/extensibles/content/PrivateTab.jsm

```diff
@@ -400,19 +400,25 @@ const PrivateTab = {
   },
 
   togglePrivate(aWindow, aTab = aWindow.gBrowser.selectedTab) {
+    let newTab;
     let { gBrowser } = aWindow;
     aTab.setAttribute("isToggling", true);
     let shouldSelect = aTab == aWindow.gBrowser.selectedTab;
-    let newTab = gBrowser.duplicateTab(aTab);
-    if (shouldSelect) {
-      let gURLBar = aWindow.gURLBar;
-      let focusUrlbar = gURLBar.focused;
-      gBrowser.selectedTab = newTab;
-      if (focusUrlbar) {
-        gURLBar.focus();
+    try {
+      newTab = gBrowser.duplicateTab(aTab);
+      if (shouldSelect) {
+        let gURLBar = aWindow.gURLBar;
+        let focusUrlbar = gURLBar.focused;
+        gBrowser.selectedTab = newTab;
+        if (focusUrlbar) {
+          gURLBar.focus();
+        }
       }
+      gBrowser.removeTab(aTab);
+    } catch (ex) {
+      // Can use this to pop up failure message
     }
-    gBrowser.removeTab(aTab);
+    return newTab;
```

Comment:
Is returning `undefined` ok? It seems the changes add this possibility.

Reply:
The return value is only used in some tests as we need to have access to the new tab to be able to remove it at the end of the test. Line 7 in `browser/extensions/extensibles/test/browser/browser_privatetab.js`. I added the try catch block as one of the tests was occasionally failing to find `gBrowser.selectedTab`, if memory serves.
For the general running of the browser the return value would never be used.